pkg/operator/sync.go

```diff
@@ -30,22 +29,32 @@ type syncFunc struct {
 	fn   func(config renderConfig) error
 }
 
+type syncError struct {
+	task string
+	err  error
+}
+
 func (optr *Operator) syncAll(rconfig renderConfig, syncFuncs []syncFunc) error {
 	if err := optr.syncProgressingStatus(); err != nil {
 		return fmt.Errorf("error syncing progressing status: %v", err)
 	}
 
-	var errs []error
+	var syncErr syncError
 	for _, sf := range syncFuncs {
 		startTime := time.Now()
-		errs = append(errs, sf.fn(rconfig))
+		syncErr = syncError{
+			task: sf.name,
+			err:  sf.fn(rconfig),
+		}
```

Comment:
in 4.1 this is going to shadow errors coming from the other sync functions as we're not breaking at the first error like in 4.2 - this needs another angle to tackle um

Reply:
hm, I added the break, see below. Is this what you mean here?

Reply:
I think this is good actually.